executor/src/witgen/mod.rs

```diff
@@ -375,6 +383,7 @@ pub struct FixedData<'a, T: FieldElement> {
     challenges: BTreeMap<u64, T>,
     global_range_constraints: GlobalConstraints<T>,
     intermediate_definitions: BTreeMap<PolyID, &'a AlgebraicExpression<T>>,
+    stage: u8,
```

Comment:
Now it's only fixed for one stage of witness generation.

Reply:
Although we already had `challenges` right?

Reply:
Yes, it's fixed for the duration of the call to `WitnessGenerator::generate`!

Reply:
Can you mention that in the docstring? ("Fixed for witness generation for a certain stage")